### ISSUE TITLE
missing exit record for "main" in i386

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#762]
Here is the i386 build binary.
```
$ gcc -m32 -pg tests/s-abc.c

$ file a.out 
a.out: ELF 32-bit LSB executable, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-, for GNU/Linux 2.6.32, BuildID[sha1]=aac6b820e3cf314a929de234038e25b79420af04, not stripped
```
It doesn't have exit record of `main` function as follows:
```
$ uftrace a.out 
# DURATION     TID     FUNCTION
   1.130 us [ 19433] | __monstartup();
   1.070 us [ 19433] | __cxa_atexit();
            [ 19433] | main() {
            [ 19433] |   a() {
            [ 19433] |     b() {
            [ 19433] |       c() {
   0.767 us [ 19433] |         getpid();
   1.997 us [ 19433] |       } /* c */
   3.154 us [ 19433] |     } /* b */
   4.206 us [ 19433] |   } /* a */

uftrace stopped tracing with remaining functions
================================================
task: 19433
[0] main
```

**Comment by honggyukim** [comment]
@ParkHanbum Can you please take a look at this problem?

**Comment by honggyukim** [comment]
It may be the reason of failing many tests in i386.

**Comment by ParkHanbum** [comment]
@honggyukim This problem occurs because of the stack align for i386. to fix this issue needs specially rule for main function. but I don't know that this is good or not.

**Comment by honggyukim** [comment]
Thanks but I don't understand why it needs a special rule for main handling.

**Comment by ParkHanbum** [comment]
refer this: 
https://github.com/namhyung/uftrace/blob/aa1f43a0dfe993da5b5a912359365d26c86193d7/arch/i386/mcount-support.c#L187

main function prolog do more stuff in i386. as you can see, it manipulate stack for stack align. so, we cannot use same way like before to hooking the return address of main function.

**Comment by ParkHanbum** [comment]
but there is something wrong. I create new issue about this.

**Comment by honggyukim** [comment]
It's fixed by #806. Thanks @ParkHanbum for fixing this.
